cmds/boot/boot.go

```diff
@@ -0,0 +1,451 @@
+// Copyright 2012-2017 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// boot allows to handover a system running linuxboot/u-root
+// to a legacy preinstalled operating system by replacing the traditional
+// bootloader path
+
+//
+// Synopsis:
+//	boot
+//
+// Description:
+//	If returns to u-root shell, the code didn't found a local bootable option
+//
+// Notes:
+//	The code is looking for boot/grub/grub.cfg file as to identify the
+//	boot option.
+//	The first bootable device found in the block device tree is the one used
+//	Windows is not supported (that is a work in progress)
+//
+// Example:
+//	boot -v 	- Start the script in verbose mode for debugging purpose
+
+package main
+
+import (
+	"bufio"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"github.com/u-root/u-root/pkg/kexec"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+	"flag"
+	"syscall"
+)
+
+const (
+	bootableMBR = 0xaa55
+	signatureOffset = 510
+)
+
+var verbose bool
+type options struct {
+        verbose bool
+}
+
+func blkDevicesList(blkpath string, devpath string) ([]string,error) {
+	var blkDevices []string
+	files, err := ioutil.ReadDir(blkpath)
+	if err != nil {
+		return blkDevices,err
+	}
+	for _, file := range files {
+		check,err := os.Stat(blkpath + file.Name() + devpath)
+		if ( check == nil ) {
+                        continue
+                }
+                if ( err != nil ) {
+                        continue
+                }
+		deviceEntry, err := ioutil.ReadDir(blkpath + file.Name() + devpath)
+		if err != nil {
+			if ( verbose ) {
+				log.Printf("can t read directory")
+			}
+			continue
+		}
+		blkDevices = append(blkDevices, deviceEntry[0].Name())
+	}
+	return blkDevices,nil
+}
+
+// checkForBootableMBR is looking for bootable MBR signature 
+// Current support is limited to Hard disk devices and USB devices
+func checkForBootableMBR(path string) error {
+	var sig uint16
```

Comment:
A common rule in go is to avoid declaring things before you use them. So, one option is to move the sig declaration to right before the binary.Read